File: Lab2/src/main/scala/Coffee/Cappuccino.scala
package Coffee

class Cappuccino(intensity: Intensity, mlOfWater: Int, val mlOfMilk: Int)
  extends Americano(intensity, mlOfWater):

  private val coffeeName: String = "Cappuccino"

  def getCoffee: String = coffeeName

  override def printCoffeeDetails(): Unit =
    println(s"$coffeeName water: ${getMlOfWater} ml, milk: $mlOfMilk ml")

  // Base method
  def makeRecipe(): Unit =
    println(s"Intensity set to $coffeeIntensity")
    println(s"Adding $mlOfMilk ml of milk")

  // Specialized method
  def makeCappuccino(): Cappuccino =
    println(s"Making $coffeeName")
    this.makeRecipe()
    this

File: lab2/src/main/kotlin/labo2/Task3/Cappuccino.kt
package org.example.labo2.Task3

import org.example.labo2.Task1.Intensity

open class Cappuccino(
    private var intensity: Intensity,
    private var mlOfMilk: Int,
    private val coffee: String = "Cappuccino"
) : Coffee(intensity, coffee) {
    // Inheriting, overriding printCoffeeDetails from Coffee and making it a base class
    public override fun printCoffeeDetails() {
        super.printCoffeeDetails()
        println("Milk: $mlOfMilk ml")
    }
    // Method to print recipe of Cappuccino
    public fun makeCappuccino(): Cappuccino {
        super.makeCoffee()
        println("Adding $mlOfMilk ml of milk")
        return this
    }
}